src/app/beer_garden/api/http/handlers/v1/user.py

```diff
@@ -80,38 +55,31 @@ def delete(self, user_id):
         tags:
           - Users
         """
-        principal = Principal.objects.get(id=str(user_id))
-        principal.delete()
+        user = User.objects.get(username=username)
+        user.delete()
```

Comment:
I think the answer in both cases is no, those exceptions should not be handled either here or in the error map.  The short version of why is that both of the conditions you describe sound like failures on the server end where the appropriate response would be a 500, which is the behavior you get when not handling the exceptions.

To take one of the examples, the only way to get `MultipleObjectsReturned` would be if someone had manually mucked with the DB indexes to remove the unique constraint, and then inserted a duplicate. And if that happened and we get a `MultipleObjectsReturned` here as a result, what's the appropriate HTTP response code to throw back?  I believe it's a 500, since it's completely a failure of the application (if in the future we accidentally removed the unique index for example) or the application administrator (if they purposefully removed said index) and nothing the user could change in a subsequent request would change anything.

Reply:
Fair enough.